duration.go

```diff
@@ -122,33 +137,101 @@ func Parse(d string) (*Duration, error) {
 	return duration, nil
 }
 
-// ToTimeDuration converts the *Duration to the standard library's time.Duration
-// note that for *Duration's with period values of a month or year that the duration becomes a bit fuzzy
+// FromTimeDuration converts the given time.Duration into duration.Duration.
+// Note that for *Duration's with period values of a month or year that the duration becomes a bit fuzzy
+// since obviously those things vary month to month and year to year
+func FromTimeDuration(d time.Duration) (*Duration, error) {
+	if d < 0 {
```

Comment:
I'm okay with supporting negatives since it's a part of the extension. Apologies for the confusion.

Reply:
I have added negative support, could you check?